demo/demos/kuramoto_model/kuramoto_model.gd

```diff
@@ -84,29 +86,37 @@ func _restart_simulation() -> void:
 	solver.initialize()
 
 func _generate_fireflies() -> void:
-	var sprite: Sprite2D
-	var num_fireflies: int = fireflies.get_child_count()
+	TEXTURE_SIZE = int(ceil(sqrt(N)))
```

Comment:
Are you sure we can't just make the texture 1D? Then we won't need `var pixel_pos := Vector2(int(i / TEXTURE_SIZE), int(i % TEXTURE_SIZE))` later either. Or worst case just 1xN (2D but just 1 pixel high)

Reply:
Ah yeah, could do 1XN. I just followed the usual convention of 2Ding it.. I don't really know if there's a significant performance boost because of it.

Reply:
I'd be surprised if there was a difference in performance between the two, except the final texture having slightly fewer pixels total in the 1xN version. It's all buffers sooner or later.

Reply:
On a quick reading, performance seems more or less same, with 1D being marginally better. But there is a limit to the size of the 1D texture, although we probably don't hit that limit in this demo. For the sake of readability, I suppose I can refactor it to 1D.

Reply:
There is hard limit of 16384 px x 16384 px on textures even on modern hardware. Since this demo can easily push past 20k entities, it makes sense to leave it as it is.